idunn/api/poi.py

```diff
@@ -0,0 +1,53 @@
+from apistar import types, validators
+from idunn.blocks.phone import PhoneBlock
+from idunn.blocks.opening_hour import OpeningHourBlock
+
+BLOCKS_ORDER = [
+    OpeningHourBlock,
+    PhoneBlock
+]
+
+def build_blocks(es_poi, lang):
+    return [c.from_es(es_poi, lang) for c in BLOCKS_ORDER]
+
+class POI(types.Type):
+    id = validators.String(allow_null=True)
+    name = validators.String(allow_null=True)
+    local_name = validators.String(allow_null=True)
+    class_name = validators.String(allow_null=True)
+    subclass_name = validators.String(allow_null=True)
+    geometry = validators.Object(allow_null=True)
+    address = validators.Object(allow_null=True)
+    blocks = validators.Array(items=BLOCKS_ORDER)
+
+    @classmethod
+    def load_poi(cls, es_poi, lang):
+        geom = None
+        if 'coord' in es_poi:
+            coord = es_poi.get('coord')
+            if 'lon' in coord and 'lat' in coord:
+                lon = coord.get('lon')
+                lat = coord.get('lat')
+                geom = {
+                    'coordinates': [lon, lat],
+                    'type': 'Point'
+                }
+
+        local_name = es_poi.get('properties', {}).get('name')
+        if lang is None:
+            name = local_name
+        else:
+            name = es_poi.get('properties', {}).get(f'name:{lang}')
```

Comment:
it does not work if the lang is not in the name.

maybe you should wrap this in a function, this way you can add some doc test on the corner cases (and some comments)